clang/test/Sema/warn-lifetime-analysis-capture-by.cpp

```diff
@@ -0,0 +1,220 @@
+// RUN: %clang_cc1 --std=c++20 -fsyntax-only -Wdangling -Wdangling-field -Wreturn-stack-address -verify %s
+
+#include "Inputs/lifetime-analysis.h"
```

Comment:
Perhaps let's split the test file that checks the capture-by on the standard library from a test file that checks the capture-by in a self contained way (without the include that imitates the standard library)?

Reply:
I don't think that would be useful. Both lifetimebound and lifetime_capture share the semantics of STL types specially the inferred lifetimebound and gsl::Pointer/Owner. 

Also, I think it is useful to test how the stl types interact with other use defined types.

We often duplicate tests for both stl types and user defined types and I would prefer to keep the redundancy together in one place as compared to repeating similar tests to two different files.

Reply:
In my experience, a single and well-organized test file works effectively, as it simplifies adding new test cases.
Having multiple files can add overhead, requiring extra effort to determine the right file for each new case. I generally prefer a workflow where new test cases are appended to one main file.

> I think it would be much more readable if we would have a namespace per use case or set of use cases where everything relevant to that use case would be declared/defined and used.

+1, I think the test can be improved by having a dedicated namespace per use case.

Reply:
> > I think it would be much more readable if we would have a namespace per use case or set of use cases where everything relevant to that use case would be declared/defined and used.
> 
> +1, I think the test can be improved by having a dedicated namespace per use case.

Done.
I will also incline towards not splitting it the test further.
Happy to address more suggestions to make this readable and organised.